File: backend/schema/llm_chat_schema.py
from pydantic import BaseModel
from typing import List

class LLMRequest(BaseModel):
    user_message: str
    model:str = "llama-3.3-70b-versatile"
    created_at: str
    user_id: str

class LLMResponse(BaseModel):
    summary: str
    title: str | None = None
    topics: List[str] #3 key topics
    sentiment: str #positive / neutral / negative
    keywords: List[str] = [] # 3 most frequent nouns

    created_at: str


class LLMChat(BaseModel):
    id: str
    llm_request: LLMRequest
    llm_response: LLMResponse

File: web/src/types/llmChatSchema.ts
export type LLMRequest = {
  user_message: string;
  model?: string; // default: "llama-3.3-70b-versatile"
  created_at: string;
  user_id: string;
};

export type LLMResponse = {
  summary: string;
  title?: string | null;
  topics: string[]; // 3 key topics
  sentiment: "positive" | "neutral" | "negative";
  keywords?: string[]; // 3 most frequent nouns
  created_at: string;
};

export type LLMChat = {
  id: string;
  llm_request: LLMRequest;
  llm_response: LLMResponse;
};
